### PR TITLE
Add support for  and bypasses a write lock

### LINKED ISSUE #3077: [BUILDER] - Support batch submit for private mempool
### What is this task and why do we need to work on it?

Batch submits will be more efficient for rollups

### What work will need to be done to complete this task?

New API endpoint in submit.toml, update to AcceptsTxnSubmissions trait, change in hotshot-builder-core

### Are there any other details to include?

_No response_

### What are the acceptance criteria to close this issue?

API changes added; because simple builder doesn't use the private mempool, no work needed there.

### Branch work will be merged to (if not the default branch)

_No response_

### LINKED ISSUE #3078: [BUILDER] - Builder Core support: Make submit_txn not require write lock
### What is this task and why do we need to work on it?

Write lock is not required for our implementation, and adds concurrency contention to the web service layer.

### What work will need to be done to complete this task?

_No response_

### Are there any other details to include?

_No response_

### What are the acceptance criteria to close this issue?

Updates to builder APIs

### Branch work will be merged to (if not the default branch)

_No response_

### PR DESCRIPTION
**Pull request by nyospe**
Closes #3077
Closes #3078

### This PR: 
Updates the builder API to support batching for private mempool submits
Updates the builder API to remove mandatory-write-locking for submit API trait.

### CODE REVIEW

File: crates/builder-api/api/submit.toml [hunk 1]
```diff
@@ -29,5 +29,10 @@ FORMAT_VERSION = "0.1.0"
 
 [route.submit_txn]
 PATH = ["/submit"]
-METHOD = "POST"
+METHOD = "AT"
```
Comment (jbearer) [inline crates/builder-api/api/submit.toml:32]: This stays `POST`

File: crates/builder-api/api/submit.toml [hunk 2]
```diff
@@ -29,5 +29,10 @@ FORMAT_VERSION = "0.1.0"
...
 DOC = "Submit a transaction to builder's private mempool."
+
+[route.submit_batch]
+PATH = ["/batch"]
+METHOD = "AT"
```
Comment (jbearer) [inline crates/builder-api/api/submit.toml:37]: `POST`

Comment (nyospe) [reply crates/builder-api/api/submit.toml:37]: So, even if it is AT internally?

**Review by move47 — APPROVED** [review]
One test seems to be failing in CI

**Review by move47 — APPROVED** [review]

**Pull request merged**